internal/expr/query.go

```diff
@@ -12,35 +13,53 @@ func (qe *QueryExpr) QueryArgs() []any {
 	return qe.args
 }
 
+func (qe *QueryExpr) SQL() string {
+	return qe.sql
+}
+
 func (qe *QueryExpr) HasOutputs() bool {
 	return len(qe.outputs) > 0
 }
 
+// QueryExpr represents a complete SQLair query, ready for execution on a
+// database.
 type QueryExpr struct {
 	sql     string
 	args    []any
 	outputs []typeMember
 }
 
-// Query returns a query expression ready for execution, using the provided values to
+const markerPrefix = "_sqlair_"
+
+func markerName(n int) string {
+	return markerPrefix + strconv.Itoa(n)
+}
+
+// markerIndex returns the int X from the string "_sqlair_X".
+func markerIndex(s string) (int, bool) {
+	if strings.HasPrefix(s, markerPrefix) {
+		n, err := strconv.Atoi(s[len(markerPrefix):])
+		if err == nil {
+			return n, true
+		}
+	}
+	return 0, false
+}
```

Comment:
When we retrieve the column names from the results use `markerIndex` to parse them. The int returned is used to find the struct field/map key that we want to scan the result in that column into. It would be good to remember how the columns relate to the `typeMember` values simply by the position/index of the column but then if a non-SQLair column was returned in the query it would mess up our system.

We could store the `typeMember` values currently stored in `qe.outputs` in a map indexed by the column names or something along those lines, which may make the code a little simpler, but I think you're right that it would be best in a separate PR.

Reply:
Reading your later comments, I think it would be great to store the `typeMember` values and column names together which would eliminate the need for this function. But I think it would be best brought as a subsequent PR since this one is already rather large.